native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java

```diff
@@ -0,0 +1,298 @@
+
+public class BallerinaDestinationDataProvider implements DestinationDataProvider {
+
+    private final Map<String, Properties> destinationProperties = new HashMap<>();
+
+    @Override
+    public Properties getDestinationProperties(String destinationName) {
+        if (destinationProperties.containsKey(destinationName)) {
+            return destinationProperties.get(destinationName);
+        } else {
+            throw new RuntimeException("Destination " + destinationName + " not found");
+        }
+    }
+
+    @Override
+    public void setDestinationDataEventListener(DestinationDataEventListener eventListener) {
+    }
+
+    @Override
+    public boolean supportsEvents() {
+        return true;
+    }
+
+    public void addDestination(BMap<BString, Object> jcoDestinationConfig) {
+        Properties properties = new Properties();
+        properties.setProperty(DestinationDataProvider.JCO_CLIENT, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("jcoClient")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_USER, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("user")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_PASSWD, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("password")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_LANG, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("language")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_ASHOST, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("host")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_SYSNR, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("systemNumber")).toString());
+        properties.setProperty(DestinationDataProvider.JCO_GROUP, jcoDestinationConfig.getStringValue(
+                StringUtils.fromString("group")).toString());
+        if (jcoDestinationConfig.getStringValue(StringUtils.fromString("authType")) != null) {
+            properties.setProperty(DestinationDataProvider.JCO_AUTH_TYPE, jcoDestinationConfig.getStringValue(
+                    StringUtils.fromString("authType")).toString());
+        }
+        if (jcoDestinationConfig.getStringValue(StringUtils.fromString("codePage")) != null) {
+            properties.setProperty(DestinationDataProvider.JCO_CODEPAGE, jcoDestinationConfig.getStringValue(
+                    StringUtils.fromString("codePage")).toString());
+        }
+        if (jcoDestinationConfig.getStringValue(StringUtils.fromString("aliasUser")) != null) {
+            properties.setProperty(DestinationDataProvider.JCO_ALIAS_USER, jcoDestinationConfig.getStringValue(
+                    StringUtils.fromString("aliasUser")).toString());
+        }
+        if (jcoDestinationConfig.getStringValue(StringUtils.fromString("pcs")) != null) {
```

Comment:
Lets check if we can use match clause

Reply:
Optimized with [1587866](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/2/commits/1587866408961566b051f449be7fd7303ce85a9f).